src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceShareService.java

```diff
@@ -1,39 +1,95 @@
+
+    private final SscsCaseCallbackDeserializer sscsCaseCallbackDeserializer;
+
+    private final DocumentManagementService documentManagementService;
+
+    private final DocumentRequestFactory documentRequestFactory;
+
+    private final EvidenceManagementService evidenceManagementService;
+
+    private final BulkPrintService bulkPrintService;
 
     @Autowired
-    private DocumentRequestFactory documentRequestFactory;
+    public EvidenceShareService(
+        SscsCaseCallbackDeserializer sscsCaseCallbackDeserializer,
+        DocumentManagementService documentManagementService,
+        DocumentRequestFactory documentRequestFactory,
+        EvidenceManagementService evidenceManagementService,
+        BulkPrintService bulkPrintService
+    ) {
+
+        this.sscsCaseCallbackDeserializer = sscsCaseCallbackDeserializer;
+        this.documentManagementService = documentManagementService;
+        this.documentRequestFactory = documentRequestFactory;
+        this.evidenceManagementService = evidenceManagementService;
+        this.bulkPrintService = bulkPrintService;
+    }
 
     public long processMessage(final String message) {
         Callback<SscsCaseData> sscsCaseDataCallback = sscsCaseCallbackDeserializer.deserialize(message);
 
         log.info("Processing callback event {} for case id {}", sscsCaseDataCallback.getEvent(),
             sscsCaseDataCallback.getCaseDetails().getId());
 
-        DocumentHolder holder = documentRequestFactory.create(sscsCaseDataCallback.getCaseDetails().getCaseData());
+        final SscsCaseData caseData = sscsCaseDataCallback.getCaseDetails().getCaseData();
+        DocumentHolder holder = documentRequestFactory.create(caseData);
 
         if (holder.getTemplate() != null) {
-            documentManagementService.generateDocumentAndAddToCcd(holder, sscsCaseDataCallback.getCaseDetails().getCaseData());
+            Pdf newPdfAddedToTheCase = documentManagementService.generateDocumentAndAddToCcd(holder, caseData);
+            List<Pdf> existingCasePdfs = toPdf(caseData.getSscsDocument());
+            bulkPrintService.sendToBulkPrint(getAllCasePdfs(newPdfAddedToTheCase, existingCasePdfs), caseData);
         }
 
         return sscsCaseDataCallback.getCaseDetails().getId();
     }
+
+    private List<Pdf> getAllCasePdfs(Pdf newPdfAddedToTheCase, List<Pdf> existingCasePdfs) {
+        List<Pdf> allPdfs = new ArrayList<>(existingCasePdfs);
+        allPdfs.add(0, newPdfAddedToTheCase);
+        return allPdfs;
+    }
+
+    private List<Pdf> toPdf(List<SscsDocument> sscsDocument) {
+        if (sscsDocument == null) {
+            return Collections.emptyList();
+        }
+        return sscsDocument.stream().flatMap(f ->
+            f.getValue().getDocumentFileName().equalsIgnoreCase("pdf")
+                ? Stream.of(new Pdf(toBytes(f), f.getValue().getDocumentFileName()))
+                : Stream.empty()
+        ).collect(Collectors.toList());
```

Comment:
Review if it's correct to assume the file name has a pdf file extension.

Reply:
I believe this is partly correct. We need to use Docmosis to stitch together all evidence into a single PDF, and then send this PDF to Bulk Print. So I guess as part of the stitching work we can refactor this method to only send this pdf.